src/utils/weights.rs

```diff
@@ -53,8 +57,8 @@ impl WeightInit {
         Value::from(weight)
     }
 
-    fn he_uniform(&self, fan_num: usize) -> Value {
-        let limit = (3.0 / (fan_num as f64)).sqrt();
+    fn he_uniform(&self, fan_in: usize) -> Value {
+        let limit = (6.0 / (fan_in as f64)).sqrt();
```

Comment:
@jrego43 Looks like we have our answer, we both missed it by Pytorch calculates a gain depending on the activation function used. I only noticed this today so having the $\sqrt{6}$ and $\sqrt{2}$ might be a generalization, thanks for catching this Demba